docs/multiple_image_support_design_proposal.md

```diff
@@ -0,0 +1,549 @@
+fragmentation- Feature Name: Multiple Image Support
+- Start Date: 2018-07-06
+- RFC PR:
+
+# Summary
+Firmware for ARMv8-M MCUs are often released in two separate binary images:
+  - secure code
+  - non-secure code
+
+Currently MCUBoot can only handle single firmware image. MCUBoot could be
+extended to handle the secure and non-secure binaries in one single blob and do
+the upgrade of the images in one round.
+This single blob approach has some drawbacks:
+  - It always needs to update both images even if one of them has not changed.
```

Comment:
This isn't that much of an issue if the upgrade process uses delta compression.

Reply:
How this delta upgrade works in general? Does the update client copy the active image to slot_1 and then apply the downloaded diffs on that to create the new image? Is delta upgrade approach widely used? It gives extra complexity to the update client, but on the other hand reduce the size of the downloaded image...

Reply:
Does MCUBoot support delta compression? is it in the road map?

Reply:
It is not on the roadmap. So how does delta compression work if the upgrade is interrupted halfway through?

Reply:
As far as I can understand you don't start the upgrade before you have received the full delta and when you do the delta upgrade you store the progress successively. 
![dfu_delta 1](https://user-images.githubusercontent.com/3646850/46010351-98876a80-c0c2-11e8-83a2-aecb52f445ff.png)

Reply:
Ok, let me rephrase it a bit... suppose you are going to upgrade one sector and you have the delta. You read the sector to RAM, apply the delta changes in RAM, and now you erase the sector because you want to write the upgrade data. Now for some reason the device resets, and when you reboot you have the erased sector and some delta to something that does not exist anymore. I suppose you could write the applied delta changes to scratch as well, as well as metadata to know which sector is being upraded,etc, so it might be feasible but might be a somewhat less straight forward than it might seem at first!